### ISSUE TITLE
build glibc into appimage

### ISSUE THREAD

**Issue by xaljer** [helix-editor/helix#11103]
I want to try helix on Ubuntu 20.04, but glibc version is too low. It would be better if providing a self contained version of appimage.

**Comment by kirawi** [comment]
Related https://github.com/helix-editor/helix/issues/1932

**Comment by adamperkowski** [comment]
You can build the appimage yourself using tools like [this](https://github.com/stratusfearme21/cargo-appimage) for example.

**Comment by Samueru-sama** [comment]
There is this [appimagetool ](https://github.com/probonopd/go-appimage) which has a deploy everything mode that bundles glibc in the appimage that makes the appimage be able to run in any distro, be it very old distros or even musl distros. It also uses a static runtime which prevents the no libfuse2 issues in newer distros.

I just tested converting the existing appimage by extracting it and then running `./appimagetool -s deploy squashfs-root/usr/share/applications/Helix.desktop` followed by `ARCH=x86_64 VERSION=test ./appimagetool -s squashfs-root` and it works, it is possible that deploying everything might introduce a bug though.

Ideally you would build with a statically link musl and then deploy everything.

**Comment by rahji** [comment]
Yes, can we please get a binary in the Releases page that would allow me to just download it and run it on any old linux system? I'm on an unfamiliar machine right now without many permissions. I'd love to just wget the hx binary and start using it but I'm getting the message that glibc is too old. I've been totally spoiled by Go and assume everything can just be a single binary that just works. 😭

**Comment by Samueru-sama** [comment]
I just made what I think are the [necessary changes](https://github.com/helix-editor/helix/compare/master...Samueru-sama:helix:master)

However how do you start the CI here so that I can verify that everything builds correctly? I don't want to make the PR without testing

**Comment by kirawi** [comment]
I'm not sure we want this. https://github.com/helix-editor/helix/issues/1932#issuecomment-1572155314

**Comment by Samueru-sama** [comment]
> I'm not sure we want this. [#1932 (comment)](https://github.com/helix-editor/helix/issues/1932#issuecomment-1572155314)
> 
> But this up to a maintainer to review to decide if that still stands.

In this case you are not actually supporting a old version of glibc, you are simply bundling the glibc of the CI into the appimage, which for all purpose has the same result as the appimage being able to run in very old distributions and has the plus that it can also run in musl distributions as well. 

I already linked the diff above showing the needed changes, in any case you don't want to maintain that let me know so I can setup a repo that builds set appimage for the people that need it.

**Comment by Samueru-sama** [comment]
Artifact for testing: https://github.com/helix-editor/helix/actions/runs/10031937785/artifacts/1726338025

The appimage with a bundled glibc is the one that contains `anylinux` in the zip file. 

If your distribution has `/bin` and `/usr/bin` separated (which is common on very old distros) you might need to launch the appimage with `FUSERMOUNT_PROG=/bin/fusermount ./Helix-dev-anylinux-x86_64.AppImage` or just make a symlink to fusermount in `/usr/bin` if you have sudo access.

Oddly enough the size increase ended up being less than 1 MiB with bundled glibc in the anylinux appimage.

**Comment by hholst80** [comment]
Just build helix with [muslc](https://wiki.musl-libc.org/) and it will avoid any external dependencies to glibc
